TeamSnapSDK/SDK/DataTypes/TSDKTeam.m

```diff
@@ -280,13 +280,7 @@ - (NSURL * _Nullable)teamPhotoForWidth:(NSInteger)width height:(NSInteger)height
 }
 
 - (id)copyWithZone:(nullable NSZone *)zone {
-    id copy = [[[self class] allocWithZone:zone] init];
-    
-    if (copy) {
-        [copy setCollection:[[self collection] copy]];
-    }
-    
-    return copy;
+    return [[[self class] allocWithZone:zone] initWithCollection:self.collection];
```

Comment:
I don't think this creates a deep copy.. The dictionary will be a new object, but the items in the dictionary wont be. 

`[[[self class] allocWithZone:zone] initWithCollection:[self.collection copy]];` ? 

but my bigger question is why does this object need a special `copyWithZone`? CollectionObject already has that. If we don't need anything that's not coming from the collection we can either remove this or `return [super copyWithZone:]`?  (the history isn't clear).

Reply:
(I have the same concern about the lack of a deep copy on TSDKCollectionObject copyWithZone)

Reply:
Never mind.. I noticed, noted, and forgot collection was copy. 

Sorry.